app/backend/src/couchers/servicers/requests.py

```diff
@@ -137,33 +140,30 @@ def CreateHostRequest(self, request, context):
 
     def GetHostRequest(self, request, context):
         with session_scope() as session:
-            host_request = (
-                session.query(HostRequest)
-                .filter_users_column(context, HostRequest.from_user_id)
-                .filter_users_column(context, HostRequest.to_user_id)
-                .filter(HostRequest.conversation_id == request.host_request_id)
-                .filter(or_(HostRequest.from_user_id == context.user_id, HostRequest.to_user_id == context.user_id))
-                .one_or_none()
-            )
+            host_request = session.execute(
+                select(HostRequest)
+                .where_users_column_visible(context, HostRequest.from_user_id)
+                .where_users_column_visible(context, HostRequest.to_user_id)
+                .where(HostRequest.conversation_id == request.host_request_id)
+                .where(or_(HostRequest.from_user_id == context.user_id, HostRequest.to_user_id == context.user_id))
+            ).scalar_one_or_none()
 
             if not host_request:
                 context.abort(grpc.StatusCode.NOT_FOUND, errors.HOST_REQUEST_NOT_FOUND)
 
-            initial_message = (
-                session.query(Message.time)
-                .filter(Message.conversation_id == host_request.conversation_id)
+            initial_message = session.execute(
+                select(Message)
+                .where(Message.conversation_id == host_request.conversation_id)
                 .order_by(Message.id.asc())
                 .limit(1)
-                .one()
-            )
+            ).scalar_one()
```

Comment:
You can refactor these to user `first`, but not critical

Reply:
`scalar_one` raises an exception if more than one is detected (or none), same as `one`. I didn't really do any reimagining of queries, just translation. Happy to change this and any others to `.first` if that's more fitting

Reply:
That's fine

Reply:
I didn't realise that difference!